pkg/sitewise/datasource.go

```diff
@@ -23,13 +25,56 @@ type Datasource struct {
 }
 
 func NewDatasource(settings backend.DataSourceInstanceSettings) (*Datasource, error) {
-	cfg := client.AWSSiteWiseDataSourceSetting{}
+	cfg := models.AWSSiteWiseDataSourceSetting{}
+
 	err := cfg.Load(settings)
 	if err != nil {
 		return nil, err
 	}
-	sessions := awsds.NewSessionCache()
 
+	err = cfg.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.Region == models.EDGE_REGION && cfg.EdgeAuthMode != models.EDGE_AUTH_MODE_DEFAULT {
+		// TODO: refresh session every 4h since creds expire
```

Comment:
I think comment is wrong as `waitTime = time.Until(authInfo.SessionExpiryTime)` is used?

Reply:
removing this